cve_bin_tool/util.py

```diff
@@ -150,6 +150,9 @@ def walk(self, roots=None):
                     for i in filenames
                     if self.pattern_match(i, self.pattern)
                     and not self.pattern_match(i, self.file_exclude_pattern)
+                    and not self.directory_pattern(
```

Comment:
Why are we still using directory_pattern? Is fix mentioned by me in previous PR not working?

Reply:
> Why are we still using directory_pattern? Is fix mentioned by me in the previous PR not working?

It wasn't working because there was not an '*' after the path which is to be excluded. It may contain any no. of the path. So, I thought it will be better just to match the string beginning. It won't work otherwise. **fnmatch** takes a regex type of input. If we were to exclude /home/temp/ in a directory /home/ .
We will have to append /home/temp/ with an '*'. So, that it will work for all the sub-dir as well as files inside that dir

Reply:
Yes we can easily append that by checking 
```python
if not string.endswith("*"):
    string += "*"
```
If you use fnmatch user can specify path like `/home/temp/*.jpg` this will only ignore jpg instead of whole directory. There are also other regex patterns supported by fnmatch.